src/parsers/manifest/local/parse_local_manifest.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import idGenerator from "../../../utils/id_generator";
+import {
+  IParsedAdaptation,
+  IParsedManifest,
+  IParsedPeriod,
+  IParsedRepresentation,
+} from "../types";
+import createRepresentationIndex from "./representation_index";
+import {
+  ILocalAdaptation,
+  ILocalManifest,
+  ILocalPeriod,
+  ILocalRepresentation,
+} from "./types";
+
+const generateManifestID = idGenerator();
+
+/**
+ * @param {Object} localManifest
+ * @returns {Object}
+ */
+export default function parseLocalManifest(
```

Comment:
So we do trust the interface of the given object ? Shouldn't we add a validator ?

Reply:
For me, just a simple check for "type: local" is sufficient. 

I'm not sure that checking for every single property to be present will be better here.

Either they give us something we do not understand (without the "type" key) in which case we will reject with the right error.
Either they give us a malformed local manifest and we will just throw an error because we cannot access a given property.
Either they give us what we want and all goes well.

I'm not sure that being too tolerant here is what we want.